agent-core/src/main/java/com/newrelic/agent/android/harvest/DataToken.java

```diff
@@ -60,6 +60,13 @@ public boolean isValid() {
         return accountId > 0 && agentId > 0;
     }
 
+    public static DataToken newFromJson(JsonArray jsonArray) {
+        final DataToken dataToken = new DataToken();
+        dataToken.setAccountId(jsonArray.get(0).getAsInt());
```

Comment:
Is jsonArray valid? Is there an element at position 0?

Reply:
yes, we are only using it for reading data token previously stored crash in sharedpreference.